java/core/src/main/java/com/microsoft/bond/BondedBondType.java

```diff
@@ -0,0 +1,149 @@
+
+    /**
+     * The name of the type as it appears in Bond schemas.
+     */
+    public static final String TYPE_NAME = "bonded";
+
+    private final StructBondType<TStruct> valueType;
+
+    // restrict instantiation to the current package
+    BondedBondType(StructBondType<TStruct> valueType) {
+        super(-valueType.hashCode());
+        this.valueType = valueType;
+    }
+
+    /**
+     * Retrieves the underlying value type descriptor.
+     *
+     * @return the underlying value type descriptor
+     */
+    public final BondType<TStruct> getValueType() {
+        return this.valueType;
+    }
+
+    @Override
+    public final String getName() {
+        return TYPE_NAME;
+    }
+
+    @Override
+    public final String getQualifiedName() {
+        return TYPE_NAME;
+    }
+
+    @Override
+    public final BondDataType getBondDataType() {
+        return this.valueType.getBondDataType();
+    }
+
+    @Override
+    public final Class<Bonded<TStruct>> getValueClass() {
+        // can't do direct cast
+        @SuppressWarnings("unchecked")
+        Class<Bonded<TStruct>> valueClass = (Class<Bonded<TStruct>>) (Class<?>) Bonded.class;
+        return valueClass;
+    }
+
+    @Override
+    public final Class<Bonded<TStruct>> getPrimitiveValueClass() {
+        return null;
+    }
+
+    @Override
+    public final boolean isNullableType() {
+        return false;
+    }
+
+    @Override
+    public final boolean isGenericType() {
+        return true;
+    }
+
+    @Override
+    public final BondType<?>[] getGenericTypeArguments() {
+        return new BondType<?>[]{this.valueType};
+    }
+
+    @Override
+    protected final Bonded<TStruct> newDefaultValue() {
+        return new Bonded<TStruct>(this.valueType.newDefaultValue());
+    }
+
+    @Override
+    protected final void serializeValue(SerializationContext context, Bonded<TStruct> value) throws IOException {
+        this.verifyNonNullableValueIsNotSetToNull(value);
+
+        // TODO: complete serialization story for bonded
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected final Bonded<TStruct> deserializeValue(TaggedDeserializationContext context) throws IOException {
+        TStruct value = this.valueType.deserializeValue(context);
+
+        // TODO: complete deserialization story for bonded
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected final void serializeField(
+            SerializationContext context,
+            Bonded<TStruct> value,
+            StructBondType.StructField<Bonded<TStruct>> field) throws IOException {
+        // struct fields are never omitted
+        context.writer.writeFieldBegin(BondDataType.BT_STRUCT, field.getId(), field);
+        try {
+            this.serializeValue(context, value);
+        } catch (InvalidBondDataException e) {
+            // throws
```

Comment:
It indicates termination of the code that compiler doesn't catch, so it makes it clear that the code below always throws.

---
In reply to: [123113269](https://github.com/Microsoft/bond/pull/502#discussion_r123113269) [](ancestors = 123113269)